python/ray/tune/schedulers/hb_bohb.py

```diff
@@ -0,0 +1,216 @@
+                # cur_band will always be less than s_max_1 or else filled
+                s = self._s_max_1 - len(cur_band) - 1
+                assert s >= 0, "Current band is filled!"
+                #### MAIN CHANGE HERE!
+                if self._get_r0(s) == 0:
+                    logger.info("Bracket too small - Retrying...")
+                    cur_bracket = None
+                else:
+                    retry = False
+                    cur_bracket = ContinuationBracket(self._time_attr, self._get_n0(s),
+                                          self._get_r0(s), self._max_t_attr,
+                                          self._eta, s)
+                cur_band.append(cur_bracket)
+                self._state["bracket"] = cur_bracket
+
+        self._state["bracket"].add_trial(trial)
+        self._trial_info[trial] = cur_bracket, self._state["band_idx"]
+
+    def on_trial_result(self, trial_runner, trial, result):
+        """If bracket is finished, all trials will be stopped.
+
+        If a given trial finishes and bracket iteration is not done,
+        the trial will be paused and resources will be given up.
+
+        This scheduler will not start trials but will stop trials.
+        The current running trial will not be handled,
+        as the trialrunner will be given control to handle it."""
+        from collections import Counter
+        result["hyperband_info"] = {}
+        bracket, _ = self._trial_info[trial]
+        bracket.update_trial_stats(trial, result)
+
+        if bracket.continue_trial(trial):
+            return TrialScheduler.CONTINUE
+
+        result["hyperband_info"]["budget"] = bracket._cumul_r
+
+        #### MAIN CHANGE HERE!
+        statuses = [(t, t.status) for t in bracket._live_trials]
+        if not bracket.filled() or any(
+                status != Trial.PAUSED for t, status in statuses
+                if t is not trial):
+            print(Counter([status for _, status in statuses]))
+            trial_runner._search_alg.on_pause(trial.trial_id)
+            return TrialScheduler.PAUSE
+        # import ipdb; ipdb.set_trace()
+        action = self._process_bracket(trial_runner, bracket, trial)
+        return action
+
+    def _process_bracket(self, trial_runner, bracket, trial):
+        """This is called whenever a trial makes progress.
+
+        When all live trials in the bracket have no more iterations left,
+        Trials will be successively halved. If bracket is done, all
+        non-running trials will be stopped and cleaned up,
+        and during each halving phase, bad trials will be stopped while good
+        trials will return to "PENDING"."""
+
+        action = TrialScheduler.PAUSE
+        if bracket.cur_iter_done():
+            if bracket.finished():
+                bracket.cleanup_full(trial_runner)
+                return TrialScheduler.STOP
+
+            good, bad = bracket.successive_halving(self._reward_attr)
```

Comment:
there's no more `_reward_attr`, so this should be `_metric`. `HyperBand` also takes in a `mode` from its superclass, so I think we just need to do the negation here or something.